elastic/kibana
# [Fleet] Classify index not found as file not found either

## Summary

If a file is uploaded using a previous stack release (prior to 8.10), and then upgrading to 8.10, there is an edge case where the user may see an index not found exception with HTTP 500 errors.

This is due to changing the index names in 8.10. If the user looks at a historical view of files uploaded prior to the 8.10 upgrade _and has not yet uploaded any files since upgrade_ (which would trigger index creation via templates), then the error will be seen.

This classifies the not-yet-created-index as a missing-file error. Files uploaded prior to 8.10 will not be available, and if no files have been uploaded at all, then missing file is also a reasonable error class here.

### Further Details


Screenshot of before:

![2023-09-01-084950_scrot](https://github.com/elastic/kibana/assets/315796/5f4700d3-e312-4f8e-a9ff-cc6cdf51a98e)


And after:

![2023-09-01-085600_scrot](https://github.com/elastic/kibana/assets/315796/bdd4b2d6-833a-4fc4-9caf-1a18fce99f08)




#### Kibana error log showing the change in error type.

before:

```
[ERROR][plugins.securitySolution.actionFileInfo] FleetFilesClientError: index_not_found_exception
	Root causes:
		index_not_found_exception: no such index [.fleet-fileds-fromhost-meta-endpoint]

```


after:

```
[ERROR][plugins.securitySolution.actionFileInfo] FleetFileNotFound: index_not_found_exception
	Root causes:
		index_not_found_exception: no such index [.fleet-fileds-fromhost-meta-endpoint]
    at FleetFromHostFilesClient.get (client_from_host.ts:85:15)
```

(note the difference,  **FleetFilesClientError** vs **FleetFileNotFound**. All other error messaging is retained)


#### To Reproduce / Test:

1. Start up elasticsearch and kibana,
1. Prepare mock data of connected Endpoints:
    - `node x-pack/plugins/security_solution/scripts/endpoint/endpoint_agent_emulator.js --asSuperuser --kibana http://localhost:5601 --elasticsearch http://localhost:9200`
    - Select `1` for `Load endpoints`
    - Select `1` for `Run` (the defaults are sufficient)
    - Wait for operation to complete
1. Create a file Upload record
    - visit the endpoint management page (Security > Manage > Endpoints, or `/app/security/administration/endpoints?`)
    - Choose any row in the table, select the three-dot icon in the far right column, under **Actions** and select **Respond** in the selection menu
    - Enter a file upload command: `get-file --path="/bin/ls" --comment "respond.state=success"`
1. Delete the file indices
    - `curl -u elastic:changeme -X DELETE 'http://localhost:9200/_data_stream/.fleet-fileds*'`
1. View the response console interface again, observe the error message




### For maintainers
